### PR TITLE
Fix bug with error state writing

### PR DESCRIPTION
**Pull request by thesilentg**
Errors were not being added to the state file

### CODE REVIEW

File: processor_test.go [hunk 1]
```diff
@@ -716,10 +717,14 @@ func TestProcessor_Serialization(t *testing.T) {
 		State[MyAppContext, MyOverallContext, MyJobContext]{
 			TriggerState: TRIGGER_STATE_NEW,
 			Exec: func(ctx context.Context, ac MyAppContext, oc MyOverallContext, jc MyJobContext) (MyJobContext, string, []KickRequest[MyJobContext], error) {
+				if jc.Count == 1 {
+					return jc, STATE_DONE, nil, errors.New("errored again")
```
Comment (gaffo) [inline processor_test.go:721]: Why errors again?

File: processor.go [hunk 2]
```diff
@@ -382,13 +381,20 @@ func (s *StateExec[AC, OC, JC]) Run() {
 				s.state.RateLimit.Wait(s.ctx)
 				slog.Info("LimiterAllowed", "worker", s.i, "state", s.state.TriggerState, "job", j.Id)
 			}
+			priorState := j.State
 			// Execute the job
 			rtn := Return[JC]{
-				PriorState: j.State,
+				PriorState: priorState,
```
Comment (gaffo) [inline processor.go:387]: Is there a test that catches this bug?

File: processor_test.go (hunk 1 above)
Comment (gaffo) [inline processor_test.go:721]: Test for this?

Comment (gaffo) [reply processor_test.go:721]: I see it

Comment (thesilentg) [reply processor_test.go:721]: to confirm that if multiple errors are detected, they are recorded separately and don't clobber each other. You can see this assertion lower in the test case

File: processor.go (hunk 2 above)
Comment (thesilentg) [reply processor.go:387]: The updated test case does

**Pull request merged**